### PR TITLE
Initial java Item io

### PR DESCRIPTION
**Pull request by hirdesharma**
Added main.java as main file , 
ItemManager.java for taking input from user and asking in loop for other items , 
Item.java for calculation of tax and final price also from giving the final result to user in command line , 
ItemValidator.java for error handling and validating the data type of inputs provided by the user , 
InvalidArguments.java for throwing error message it is used in Itemvalidator.java for throwing error in case of an error

### CODE REVIEW

**Comment by sankethkini** [comment]
- pr title can be more meaningful
- commits can be more meaningful

File: src/assignment1/Exceptions/InvalidArgument.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package assignment1.Exceptions;
```
Comment (sankethkini) [inline src/assignment1/Exceptions/InvalidArgument.java:1]: package name should be all small case

File: src/assignment1/Item.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package assignment1;
```
Comment (sankethkini) [inline src/assignment1/Item.java:1]: create a new package to store models

File: src/assignment1/ItemValidator.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package assignment1;
```
Comment (sankethkini) [inline src/assignment1/ItemValidator.java:1]: create new package called validators

File: src/assignment1/Item.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+public class Item {
+
+    private String name;
+    private String type;
+    private double price;
+    private int quantity;
```
Comment (sankethkini) [inline src/assignment1/Item.java:8]: create setters and getters

File: src/assignment1/Item.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    public Item(String name, String type, double price, int quantity) {
+        this.name = name;
+        this.type = type.toLowerCase();
+        this.price = price;
+        this.quantity = quantity;
+    }
+
+    private double calculateTax() {
```
Comment (sankethkini) [inline src/assignment1/Item.java:17]: move this to separate class

File: src/assignment1/Item.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        double tax = 0;
+        switch (type.toLowerCase()) {
+            case "raw":
```
Comment (sankethkini) [inline src/assignment1/Item.java:20]: create a enum for item type

File: src/assignment1/Item.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                tax = price * 0.125;
```
Comment (sankethkini) [inline src/assignment1/Item.java:21]: create constants for values such as 0.125

File: src/assignment1/ItemManager.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+                addItem = userResponse.equalsIgnoreCase("y");
+            } catch (Exception e) {
+                System.out.println("Error: " + e.getMessage());
+            }
+        }
+        scanner.close();
+    }
+
+    private static Map<String, String> parseArguments(String[] args) {
```
Comment (sankethkini) [inline src/assignment1/ItemManager.java:47]: move this to new package called utils

File: src/assignment1/Exceptions/InvalidArgument.java (hunk 1 above)
Comment (sankethkini) [reply src/assignment1/Exceptions/InvalidArgument.java:1]: changed to small case

File: src/assignment1/Item.java (hunk 2 above)
Comment (sankethkini) [reply src/assignment1/Item.java:1]: Item should be in models, ItemManager should be in service package.

File: src/assignment1/ItemManager.java (hunk 8 above)
Comment (hirdesharma) [reply src/assignment1/ItemManager.java:47]: moved it to utils with name ArgumentParserUtils

File: src/assignment1/Item.java (hunk 7 above)
Comment (hirdesharma) [reply src/assignment1/Item.java:21]: created constants for all values like 0.125, 0.02,0.10 and others

File: src/assignment1/Item.java (hunk 6 above)
Comment (hirdesharma) [reply src/assignment1/Item.java:20]: created a package enums and created a class there called ItemType for all the types of items

File: src/assignment1/Item.java (hunk 5 above)
Comment (hirdesharma) [reply src/assignment1/Item.java:17]: moved it to a separate package called services and placed it inside a class named TaxCalculatorService

File: src/assignment1/Item.java (hunk 2 above)
Comment (hirdesharma) [reply src/assignment1/Item.java:1]: created a new package called models for Item 
and a package services for ItemManager

File: src/assignment1/Item.java (hunk 4 above)
Comment (hirdesharma) [reply src/assignment1/Item.java:8]: created getter and setters in ItemValidator

Comment (sankethkini) [reply src/assignment1/Item.java]: create getters and setter for Item as well

File: src/main/java/assignment1/Main.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package main.java.assignment1;
+
+import main.java.assignment1.services.ItemManager;
+
+public class Main {
+    public static void main(String[] args) {
+        ItemManager manager = new ItemManager();
```
Comment (sankethkini) [inline src/main/java/assignment1/Main.java:7]: use DIP

File: src/main/java/assignment1/services/TaxCalculatorService.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+import main.java.assignment1.enums.ItemType;
+
+public class TaxCalculatorService {
+    private static final double BASE_TAX_RATE = 0.125;
+    private static final double MANUFACTURED_SURCHARGE_RATE = 0.02;
+    private static final double IMPORT_DUTY_RATE = 0.10;
+    private static final double SURCHARGE_LOW = 5;
+    private static final double SURCHARGE_MEDIUM = 10;
+    private static final double SURCHARGE_HIGH_RATE = 0.05;
```
Comment (sankethkini) [inline src/main/java/assignment1/services/TaxCalculatorService.java:11]: move these to new file called TaxRateConstants, which is under constants package

File: src/test/java/main/java/assignment1/exceptions/InvalidArgumentTest.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package main.java.assignment1.exceptions;
+
+class InvalidArgumentTest {
```
Comment (sankethkini) [inline src/test/java/main/java/assignment1/exceptions/InvalidArgumentTest.java:3]: remove this file

File: src/test/java/main/java/assignment1/services/TaxCalculatorServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
+package main.java.assignment1.services;
+
+import main.java.assignment1.enums.ItemType;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TaxCalculatorServiceTest {
```
Comment (sankethkini) [inline src/test/java/main/java/assignment1/services/TaxCalculatorServiceTest.java:8]: test edge cases such as null, negative numbers, zeros etc.

File: src/test/java/main/java/assignment1/utils/ArgumentParserUtilsTest.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+import main.java.assignment1.exceptions.InvalidArgument;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class ArgumentParserUtilsTest {
```
Comment (sankethkini) [inline src/test/java/main/java/assignment1/utils/ArgumentParserUtilsTest.java:11]: test empty and null

File: src/assignment1/Item.java (hunk 4 above)
Comment (hirdesharma) [reply src/assignment1/Item.java]: created getter and setter for item

File: src/test/java/main/java/assignment1/utils/ArgumentParserUtilsTest.java (hunk 13 above)
Comment (hirdesharma) [reply src/test/java/main/java/assignment1/utils/ArgumentParserUtilsTest.java:11]: added test with null and empty inputs

File: src/test/java/main/java/assignment1/services/TaxCalculatorServiceTest.java (hunk 12 above)
Comment (hirdesharma) [reply src/test/java/main/java/assignment1/services/TaxCalculatorServiceTest.java:9]: Added test case for null , negative number and zeroes
